lio_sam:
  # Topics
  pointCloudTopic: "livox/lidar" #"/os_cloud_node/points" #"velodyne_points" #               # Point cloud data
  imuTopic: livox/imu #"/os_cloud_node/imu" #"imu_data" #                        # IMU data
  odomTopic: "odometry/imu" # IMU pre-preintegration odometry, same frequency as IMU
  gpsTopic: "odometry/gps" # GPS odometry topic from navsat, see module_navsat.launch file

  # Frames
  lidarFrame: "base_link"
  baselinkFrame: "base_link"
  odometryFrame: "odom"
  mapFrame: "map"

  # GPS Settings
  useImuHeadingInitialization: false # if using GPS data, set to "true"
  useGpsElevation: false # if GPS elevation is bad, set to "false"
  gpsCovThreshold: 2.0 # m^2, threshold for using GPS data
  poseCovThreshold: 25.0 # m^2, threshold for using GPS data

  # Export settings
  savePCD: false # https://github.com/TixiaoShan/LIO-SAM/issues/3
  savePCDDirectory: "/Downloads/LOAM/" # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

  # Sensor Settings
  sensor: livox # lidar sensor type, either 'velodyne' or 'ouster' 'robosense'
  N_SCAN: 16 # number of lidar channel (i.e., 16, 32, 64, 128)
  Horizon_SCAN: 1800 # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048)
  downsampleRate: 1 # default: 1. Downsample your data if too many points. i.e., 16 = 64 / 4, 16 = 16 / 1
  lidarMinRange: 0.2 # default: 1.0, minimum lidar range to be used
  lidarMaxRange: 1000.0 # default: 1000.0, maximum lidar range to be used

  # IMU Settings
  imuAccNoise: 1.e-2
  imuGyrNoise: 1.e-4
  imuAccBiasN: 0.3
  imuGyrBiasN: 0.04
  imuGravity: 9.80511
  imuRPYWeight: 0.0

  # Extrinsics (lidar -> IMU)
  extrinsicTrans: [0.0, -0.0, -0.0]

  extrinsicRot: [1, 0, 0, 0, 1, 0, 0, 0, 1]

  extrinsicRPY: [1, 0, 0, 0, 1, 0, 0, 0, 1]

  # LOAM feature threshold
  edgeThreshold: 1.0
  surfThreshold: 0.1
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100

  # voxel filter paprams
  odometrySurfLeafSize: 0.2 # default: 0.4 - outdoor, 0.2 - indoor
  mappingCornerLeafSize: 0.1 # default: 0.2 - outdoor, 0.1 - indoor
  mappingSurfLeafSize: 0.2 # default: 0.4 - outdoor, 0.2 - indoor

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 1000 # meters
  rotation_tollerance: 1000 # radians

  # CPU Params
  numberOfCores: 7 # number of cores for mapping optimization
  mappingProcessInterval: 0.15 # seconds, regulate mapping frequency

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 0.5 # meters, regulate keyframe adding threshold
  surroundingkeyframeAddingAngleThreshold: 0.1 # radians, regulate keyframe adding threshold
  surroundingKeyframeDensity: 0.5 # meters, downsample surrounding keyframe poses
  surroundingKeyframeSearchRadius: 20.0 # meters, within n meters scan-to-map optimization (when loop closure disabled)

  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0 # Hz, regulate loop closure constraint add frequency
  surroundingKeyframeSize: 50 # submap size (when loop closure enabled)
  historyKeyframeSearchRadius: 15.0 # meters, key frame that is within n meters from current pose will be considerd for loop closure
  historyKeyframeSearchTimeDiff: 30.0 # seconds, key frame that is n seconds older will be considered for loop closure
  historyKeyframeSearchNum: 25 # number of hostory key frames will be fused into a submap for loop closure
  historyKeyframeFitnessScore: 0.3 # icp threshold, the smaller the better alignment

  # Visualization
  globalMapVisualizationSearchRadius: 1000.0 # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0 # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 1.0 # meters, global map visualization cloud density

